test/rekt/features/broker/data_plane.go

```diff
@@ -407,3 +414,440 @@ func brokerEventVersionNotUpgraded(ctx context.Context, t feature.T) {
+	brokerName := state.GetStringOrFail(ctx, t, "brokerName")
+
+	source := feature.MakeRandomK8sName("source")
+
+	sink1 := feature.MakeRandomK8sName("sink1")
+	sink2 := feature.MakeRandomK8sName("sink2")
+	sink3 := feature.MakeRandomK8sName("sink3")
+	sink4 := feature.MakeRandomK8sName("sink4")
+
+	triggerName1 := feature.MakeRandomK8sName("trigger1")
+	triggerName2 := feature.MakeRandomK8sName("trigger2")
+	triggerName3 := feature.MakeRandomK8sName("trigger3")
+	triggerName4 := feature.MakeRandomK8sName("trigger4")
+
+	// Creating event
+	event := test.FullEvent()
+
+	// Creating sink
+	eventshub.Install(sink1, eventshub.StartReceiver)(ctx, t)
+	eventshub.Install(sink2, eventshub.StartReceiver)(ctx, t)
+	eventshub.Install(sink3, eventshub.StartReceiver)(ctx, t)
+	eventshub.Install(sink4, eventshub.StartReceiver)(ctx, t)
+
+	// Creating trigger
+	trigger.Install(triggerName1, brokerName, trigger.WithSubscriber(service.AsKReference(sink1), ""))
+	trigger.Install(triggerName2, brokerName, trigger.WithSubscriber(service.AsKReference(sink2), ""))
+	trigger.Install(triggerName3, brokerName, trigger.WithSubscriber(service.AsKReference(sink3), ""))
+	trigger.Install(triggerName4, brokerName, trigger.WithSubscriber(service.AsKReference(sink4), ""))
+
+	eventshub.Install(source,
+		eventshub.StartSenderToResource(broker.GVR(), brokerName),
+		eventshub.InputEvent(event),
+	)
+
+	assert.OnStore(sink1).MatchEvent(test.HasId(event.ID())).Exact(1)
+	assert.OnStore(sink2).MatchEvent(test.HasId(event.ID())).Exact(1)
+	assert.OnStore(sink3).MatchEvent(test.HasId(event.ID())).Exact(1)
+	assert.OnStore(sink4).MatchEvent(test.HasId(event.ID())).Exact(1)
+}
+
+func multipleTriggerSameSink(ctx context.Context, t feature.T) {
+	brokerName := state.GetStringOrFail(ctx, t, "brokerName")
+
+	source := feature.MakeRandomK8sName("source")
+	sink := feature.MakeRandomK8sName("sink")
+	triggerName1 := feature.MakeRandomK8sName("trigger1")
+	triggerName2 := feature.MakeRandomK8sName("trigger2")
+	event := test.FullEvent()
+
+	// Creating sink
+	eventshub.Install(sink, eventshub.StartReceiver)(ctx, t)
+
+	// Point the Trigger subscriber to the sink svc.
+	cfg := []manifest.CfgFn{trigger.WithSubscriber(service.AsKReference(sink), "")}
+
+	// Creating trigger
+	trigger.Install(triggerName1, brokerName, cfg...)
+	trigger.Install(triggerName2, brokerName, cfg...)
+
+	eventshub.Install(source,
+		eventshub.StartSenderToResource(broker.GVR(), brokerName),
+		eventshub.InputEvent(event),
+	)
+
+	// Since there are two triggers, there should be exactly two events
+	assert.OnStore(sink).MatchEvent(test.HasId(event.ID())).Exact(2)
+}
+
+func triggerBecameReadyLater(ctx context.Context, t feature.T) {
+	brokerName := state.GetStringOrFail(ctx, t, "brokerName")
+
+	source := feature.MakeRandomK8sName("source")
+	sink := feature.MakeRandomK8sName("sink")
+	triggerName := feature.MakeRandomK8sName("trigger")
+	event := test.FullEvent()
+
+	f := new(feature.Feature)
```

Comment:
Why are you creating a new feature (you are already in a step function)? Same question for the rest of the functions of this PR.